sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -274,6 +274,13 @@ abstract class QueryTest extends PlanTest {
     val normalized1 = logicalPlan.transformAllExpressions {
       case udf: ScalaUDF => udf.copy(function = null)
       case gen: UserDefinedGenerator => gen.copy(function = null)
+      // SPARK-17356: In usage of mllib, Metadata may store a huge vector of data, transforming
+      // it to JSON may trigger OutOfMemoryError.
+      case a @ Alias(child, name) if a.explicitMetadata.isDefined =>
+        Alias(child, name)(a.exprId, a.qualifier, Some(Metadata.empty), a.isGenerated)
+      case a: AttributeReference if a.metadata != Metadata.empty =>
+        AttributeReference(a.name, a.dataType, a.nullable, Metadata.empty)(a.exprId, a.qualifier,
+          a.isGenerated)
```

Comment:
If we have https://github.com/apache/spark/pull/14915/files#diff-eac5b02bb450a235fef5e902a2671254R623, do we still need these?